speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/ResearcherMenu.cpp

```diff
@@ -406,20 +407,20 @@ static void SelectBlackBox(void* /* dummy */)
     // Minimum file length: "{Drive Letter}:\{empty file name}.exe"
     if (path.string().size() <= 7)
     {
-        GfuiButtonSetText(s_scrHandle, m_blackBoxButton, MSG_BLACK_BOX_NOT_EXE);
+        GfuiLabelSetText(s_scrHandle, m_noBlackBoxLabel, MSG_NO_BLACK_BOX);
         return;
     }
     // Enforce that file ends in .exe
     if (std::strcmp(path.extension().string().c_str(), ".exe") != 0)
     {
-        GfuiButtonSetText(s_scrHandle, m_blackBoxButton, MSG_BLACK_BOX_NOT_EXE);
+        GfuiLabelSetText(s_scrHandle, m_noBlackBoxLabel, MSG_NO_BLACK_BOX);
```

Comment:
Why was the error message changed? I think the old message was better to reflect the error.

Reply:
This was always the error that was displayed if you tried to apply without a black box selected.
The fact that the file must match *.exe is displayed in the message hint.

Reply:
I've made it more explicit that there is a hint available on that label, as well as making sure that the messages for not having selected a black box and just choosing a file that is ineligible differ.